source/common/config/well_known_names.h

```diff
@@ -12,6 +12,8 @@
 namespace Envoy {
 namespace Config {
 
+const std::string NAME_REGEX = R"([^\.]+)";
```

Comment:
Nice idea, sharing the source of truth for what a name is, though actually it's a misnomer and this is a tag *value* not a name. I was thinking names may want a different syntax. I like the direction, though, of putting the source here.

I'd suggest doing a value check against R"(^([^\.]+)$)".

Also: no non-pod statics. You can use constexpr absl::string_view here, but not const std::string. Even then I would further refine it. Then instead of exporting a global variable I'd export a helper method to do the validation.

You can use CONSTRUCT_ON_FIRST_USE to precompile the RE2.

Then you could use separate methods to validate tag-name and tag-value.

Reply:
> though actually it's a misnomer and this is a tag value not a name.

Yeah, I noticed, I named it as `NAME_REGEX` because it replaces `<NAME>` in tag values.

sure I can switch to CONSTRUCT_ON_FIRST_USE and avoid the whole static initialization order fiasco

Reply:
resolved in 0511d8a92b40d0e3f5092199f2dd71e04d69fd04